File: invesdwin-scripting-parent/invesdwin-scripting-rust/invesdwin-scripting-rust-runtime-contract/src/test/java/de/invesdwin/scripting/rust/runtime/contract/callback/ParametersAndReturnsTestByte.rs
println!("getByte");
let getByte: i32 = callback("getByte", &[]);
println!("getByte: {}", getByte);
callback_void("setByte", &[param(getByte)]);

println!("getByteVector");
let getByteVector: Vec<i32> = callback("getByteVector", &[]);
println!("getByteVector: {:?}", getByteVector);
callback_void("setByteVector", &[param(getByteVector)]);

println!("getByteVectorAsList");
let getByteVectorAsList: Vec<i32> = callback("getByteVectorAsList", &[]);
println!("getByteVectorAsList: {:?}", getByteVectorAsList);
callback_void("setByteVectorAsList", &[param(getByteVectorAsList)]);

println!("getByteMatrix");
let getByteMatrix: Vec<Vec<i32>> = callback("getByteMatrix", &[]);
println!("getByteMatrix: {:?}", getByteMatrix);
callback_void("setByteMatrix", &[param(getByteMatrix)]);

println!("getByteMatrixAsList");
let getByteMatrixAsList: Vec<Vec<i32>> = callback("getByteMatrixAsList", &[]);
println!("getByteMatrixAsList: {:?}", getByteMatrixAsList);
callback_void("setByteMatrixAsList", &[param(getByteMatrixAsList)]);

File: invesdwin-scripting-parent/invesdwin-scripting-haskell/invesdwin-scripting-haskell-runtime-contract/src/test/java/de/invesdwin/scripting/haskell/runtime/contract/callback/ParametersAndReturnsTestByte.hs
putStrLn ( show ( "getByte" ) )
getByte :: Int <- callback "getByte" []
putStrLn ( show ( getByte ) )
callback "setByte" [param getByte] :: IO ()

putStrLn ( show ( "getByteVector" ) )
getByteVector :: [Int] <- callback "getByteVector" []
putStrLn ( show ( getByteVector ) )
callback "setByteVector" [param getByteVector] :: IO ()

putStrLn ( show ( "getByteVectorAsList" ) )
getByteVectorAsList :: [Int] <- callback "getByteVectorAsList" []
putStrLn ( show ( getByteVectorAsList ) )
callback "setByteVectorAsList" [param getByteVectorAsList] :: IO ()

putStrLn ( show ( "getByteMatrix" ) )
getByteMatrix :: [[Int]] <- callback "getByteMatrix" []
putStrLn ( show ( getByteMatrix ) )
callback "setByteMatrix" [param getByteMatrix] :: IO ()

putStrLn ( show ( "getByteMatrixAsList" ) )
getByteMatrixAsList :: [[Int]] <- callback "getByteMatrixAsList" []
putStrLn ( show ( getByteMatrixAsList ) )
callback "setByteMatrixAsList" [param getByteMatrixAsList] :: IO ()
